application-collabora-ui/src/main/resources/Collabora/Code/UI.xml

```diff
@@ -178,24 +178,23 @@ require(['jquery', 'xwiki-l10n!collabora-attachment'],function($, l10n) {
     return accessRights;
   };
 
-  var getActionURL = function(fileName, accessRights) {
-    const actionDoc = new XWiki.Document(XWiki.Model.resolve('Collabora.Code.Main', XWiki.EntityType.DOCUMENT));
+  var getActionURL = function(fileName, accessRights, pageURL) {
     const queryString = $.param({
       'document': XWiki.Document.currentDocumentReference.toString(),
       'filename': fileName,
       'action': accessRights,
       'xpage': 'plain'
     });
 
-    return actionDoc.getURL('get', queryString);
+    return pageURL + '?' + queryString;
```

Comment:
This is a bit too fragile as you can't be sure that the given URL doesn't already have a query string. I think it's worth bulletproofing this.